lightstepoc/exporter.go

```diff
@@ -0,0 +1,104 @@
+package lightstepoc
+
+import (
+	"context"
+	"errors"
+
+	"github.com/lightstep/lightstep-tracer-go"
+	"github.com/lightstep/lightstep-tracer-go/lightstepoc/internal/conversions"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/opentracing/opentracing-go/log"
+	"go.opencensus.io/trace"
+)
+
+var (
+	ErrFailedToCreateExporter = errors.New("lightstepoc: failed to create exporter")
+)
+
+type Exporter struct {
+	tracer lightstep.Tracer
+}
+
+func NewExporter(opts ...Option) (*Exporter, error) {
+	c := defaultConfig()
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	tracer := lightstep.NewTracer(c.tracerOptions)
+	if tracer == nil {
+		// Ideally we'd know *why* we couldn't create the tracer, but `NewTracer` currently doesn't return errors
```

Comment:
Lightstep Options Validate returns error https://github.com/lightstep/lightstep-tracer-go/blob/6254a61a022ea44a8e6dd828e06d40a4ea7bc0b3/options.go#L261
Which NewTraaer already takes care of, but you get the actual err by calling it yourself.